plugins/outputs/application_insights/application_insights.go

```diff
@@ -0,0 +1,422 @@
+	setFloat64TelemetryProperty(&(telemetry.StdDev), []string{"stddev"}, metric, &usedFields)
+
+	return telemetry, usedFields
+}
+
+func (a *ApplicationInsights) createTelemetryForUnusedFields(metric telegraf.Metric, usedFields []string) []appinsights.Telemetry {
+	fields := metric.Fields()
+	retval := make([]appinsights.Telemetry, 0, len(fields))
+
+	for fieldName := range fields {
+		if contains(usedFields, fieldName) {
+			continue
+		}
+
+		telemetry := a.createSimpleMetricTelemetry(metric, fieldName, true)
+		if telemetry != nil {
+			retval = append(retval, telemetry)
+		}
+	}
+
+	return retval
+}
+
+func (a *ApplicationInsights) addContextTags(metric telegraf.Metric, telemetry appinsights.Telemetry) {
+	for contextTagName, tagSourceName := range a.ContextTagSources {
+		if contextTagValue, found := metric.GetTag(tagSourceName); found {
+			telemetry.ContextTags()[contextTagName] = contextTagValue
+		}
+	}
+}
+
+func setFloat64TelemetryProperty(
+	telemetryProperty *float64,
+	candidateFields []string,
+	metric telegraf.Metric,
+	usedFields *[]string) error {
+
+	for _, fieldName := range candidateFields {
+		fieldValue, found := metric.GetField(fieldName)
+		if !found {
+			continue
+		}
+
+		metricValue, err := toFloat64(fieldValue)
+		if err != nil {
+			continue
+		}
+
+		*telemetryProperty = metricValue
+
+		if usedFields != nil {
+			*usedFields = append(*usedFields, fieldName)
+		}
+
+		return nil
+	}
+
+	return fmt.Errorf("No field from the candidate list was found in the metric")
+}
+
+func setIntTelemetryProperty(
+	telemetryProperty *int,
+	candidateFields []string,
+	metric telegraf.Metric,
+	usedFields *[]string) error {
+
+	for _, fieldName := range candidateFields {
+		fieldValue, found := metric.GetField(fieldName)
+		if !found {
+			continue
+		}
+
+		metricValue, err := toInt(fieldValue)
+		if err != nil {
+			continue
+		}
+
+		*telemetryProperty = metricValue
+
+		if usedFields != nil {
+			*usedFields = append(*usedFields, fieldName)
+		}
+
+		return nil
+	}
+
+	return fmt.Errorf("No field from the candidate list was found in the metric")
+}
+
+func contains(set []string, val string) bool {
+	for _, elem := range set {
+		if elem == val {
+			return true
+		}
+	}
+
+	return false
+}
+
+func toFloat64(value interface{}) (float64, error) {
+	switch v := value.(type) {
+	case int:
+		return float64(v), nil
+	case int8:
+		return float64(v), nil
+	case int16:
+		return float64(v), nil
+	case int32:
+		return float64(v), nil
+	case int64:
+		return float64(v), nil
+	case uint:
+		return float64(v), nil
+	case uint8:
+		return float64(v), nil
+	case uint16:
+		return float64(v), nil
+	case uint32:
+		return float64(v), nil
+	case uint64:
+		return float64(v), nil
+	case float32:
+		return float64(v), nil
+	case float64:
+		return v, nil
+	}
+
+	return 0.0, fmt.Errorf("[%s] cannot be converted to a float64 value", value)
+}
+
+func toInt(value interface{}) (int, error) {
```

Comment:
Consider returning an int64 and eliminating the 32bit cases?  Same comment as on the toFloat function, this will either be an int64, uint64, float64, bool or string.